src/zkevm_specs/evm/execution/dataCopy.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional, Sequence, Tuple, Union, List
+
+from ..instruction import Instruction, Transition
+from ..table import (
+    AccountFieldTag,
+    BlockContextFieldTag,
+    BytecodeFieldTag,
+    CallContextFieldTag,
+    FixedTableRow,
+    RWTableRow,
+    Tables,
+    FixedTableTag,
+    TxContextFieldTag,
+    RW,
+    RWTableTag,
+    TxLogFieldTag,
+    TxReceiptFieldTag,
+    CopyDataTypeTag,
+)
+from ...util import FQ, N_BYTES_MEMORY_ADDRESS, N_BYTES_MEMORY_SIZE, IdentityPerWordGas
+
+
+def dataCopy(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    call_data_offset = instruction.call_context_lookup(CallContextFieldTag.CallDataOffset, RW.Read)
+    call_data_length = instruction.call_context_lookup(CallContextFieldTag.CallDataLength, RW.Read)
+    return_data_offset = instruction.call_context_lookup(
+        CallContextFieldTag.ReturnDataOffset, RW.Read
+    )
+    return_data_length = instruction.call_context_lookup(
+        CallContextFieldTag.ReturnDataLength, RW.Read
+    )
+
+    opcall_call_id = instruction.curr.call_id
+    precompile_call_id = instruction.next.call_id
+
+    # Copy current call data to return data
+    size = call_data_length.expr()
+    copy_rwc_inc, _ = instruction.copy_lookup(
+        opcall_call_id,
+        CopyDataTypeTag.Memory,
+        opcall_call_id,
+        CopyDataTypeTag.Memory,
+        call_data_offset,
+        call_data_offset + size,
+        return_data_offset,
+        return_data_offset + return_data_length.expr(),
+        instruction.curr.rw_counter + instruction.rw_counter_offset,
+    )
+
+    # Copy current call data to next call context memory
+    copy_rwc_inc, _ = instruction.copy_lookup(
+        opcall_call_id,
+        CopyDataTypeTag.Memory,
+        precompile_call_id,
+        CopyDataTypeTag.Memory,
+        call_data_offset,
+        call_data_offset + size,
+        FQ(0),
+        return_data_length,
+        instruction.curr.rw_counter + instruction.rw_counter_offset,
```

Comment:
The `rw_counter` increment should be accumulated.

```suggestion
        instruction.curr.rw_counter + instruction.rw_counter_offset + copy_rwc_inc,
```